data/src/test/scala/io/prediction/data/api/SegmentIOAuthSpec.scala

```diff
@@ -0,0 +1,126 @@
+      reversed: Option[Boolean])
+        (implicit ec: ExecutionContext): Future[Iterator[Event]] =
+      Future successful List.empty[Event].iterator
+
+    override def futureGet(eventId: String, appId: Int, channelId: Option[Int])
+        (implicit ec: ExecutionContext): Future[Option[Event]] =
+      Future successful None
+
+    override def remove(appId: Int, channelId: Option[Int]): Boolean = true
+
+    override def futureDelete(eventId: String, appId: Int, channelId: Option[Int])
+        (implicit ec: ExecutionContext): Future[Boolean] =
+      Future successful true
+
+    override def close(): Unit = {}
+  }
+  val appId = 0
+  val accessKeysClient = new AccessKeys{
+    override def insert(k: AccessKey): Option[String] = null
+    override def getByAppid(appid: Int): Seq[AccessKey] = null
+    override def update(k: AccessKey): Unit = {}
+    override def delete(k: String): Unit = {}
+    override def getAll(): Seq[AccessKey] = null
+
+    override def get(k: String): Option[AccessKey] =
+      Some(AccessKey(k, appId, Seq.empty))
+  }
+  val channelsClient = Storage.getMetaDataChannels()
+  val eventServiceActor = system.actorOf(
+    Props(
+      new EventServiceActor(
+        eventClient,
+        accessKeysClient,
+        channelsClient,
+        EventServerConfig()
+      )
+    )
+  )
+
+  val base64Encoder = new BASE64Encoder
+
+  "Event Service" should {
+    "process SegmentIO identity request properly" in {
+      val jsonReq =
+        """
+          |{
+          |  "anonymous_id": "507f191e810c19729de860ea",
+          |  "channel": "browser",
+          |  "context": {
+          |    "ip": "8.8.8.8",
+          |    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_5)"
+          |  },
+          |  "message_id": "022bb90c-bbac-11e4-8dfc-aa07a5b093db",
+          |  "timestamp": "2015-02-23T22:28:55.387Z",
+          |  "sent_at": "2015-02-23T22:28:55.111Z",
+          |  "traits": {
+          |    "name": "Peter Gibbons",
+          |    "email": "peter@initech.com",
+          |    "plan": "premium",
+          |    "logins": 5
+          |  },
+          |  "type": "identify",
+          |  "user_id": "97980cfea0067",
+          |  "version": "2"
+          |}
+        """.stripMargin
+
+      val accessKey = "abc:"
+      val accessKeyEncoded = base64Encoder.encodeBuffer(accessKey.getBytes)
+      val probe = TestProbe()(system)
+      probe.send(
+        eventServiceActor,
+        Post(
+          "/webhooks/segmentio.json",
+          HttpEntity(ContentTypes.`application/json`, jsonReq.getBytes)
+        ).withHeaders(
+            List(
+              RawHeader("Authorization", s"Basic $accessKeyEncoded")
+            )
+          )
+      )
+      probe.expectMsg(
+        HttpResponse(
+          201,
+          HttpEntity(
+            contentType = ContentTypes.`application/json`,
+            string = """{"eventId":"event_id"}"""
+          )
+        )
+      )
+      success
+    }
+  }
```

Comment:
would you add tests for missing access key and invalid access key case as well ? (i guess can do something in the fake accessKeysClient ?)